scripts/solitaire.js

```diff
@@ -0,0 +1,797 @@
+			
+			for (var i = 0; i < dragCardsElems.length; i++) {
+				dragCardsElems[i].style.left = dragCardsElemsCoords[i].left + moveX + 'px';
+				dragCardsElems[i].style.top = dragCardsElemsCoords[i].top + moveY + 'px';
+			}
+			
+		}
+		
+	}
+
+	function findIndexByElem(obj, element) {
+		
+		for (var i = 0; i < obj.length && obj[i].elem !== element; i++);
+		return i;
+		
+	}
+
+	function placeCardBack() {
+		
+		for (var i = 0; i < dragCardsElems.length; i++) {
+			dragCardsElems[i].style.zIndex = parseInt(dragCardsElems[i].style.zIndex) - 100;
+			dragCardsElems[i].style.left = dragCardsElemsCoords[i].left + 'px';
+			dragCardsElems[i].style.top = dragCardsElemsCoords[i].top + 'px';
+		}
+		
+	}
+
+	function dropCard(e) {
+		
+		var dragedCardElem = document.elementFromPoint(e.clientX, e.clientY);
+		dragedCardElem.hidden = true;
+		
+		var elem = document.elementFromPoint(e.clientX, e.clientY);
+		
+		if ((elem.dataset.component == 'dropzone')||(elem.dataset.component == 'card')){
+			
+			if (elem.dataset.component == 'card') {
+				var currDropzone = cards[findIndexByElem(cards, elem)].placedOn;
+			} else {
+				var currDropzone = dropfields[findIndexByElem(dropfields, elem)];
+			}
+			var currCard = cards[findIndexByElem(cards, dragCardsElems[0])];
+			
+			if ((currCard.placedOn == currDropzone)||(!dropRuleCheck(currCard, currDropzone))) {
+				placeCardBack();
+				dragedCardElem.hidden = false;
+				return;
+			}
+			
+			for (var i = 0; i < dragCardsElems.length; i++) {
+				
+				currCard = cards[findIndexByElem(cards, dragCardsElems[i])];
+				placeCard(currCard, currDropzone);
+				
+			}
+			
+		} else {
+			placeCardBack();
+		}	
+		
+		dragedCardElem.hidden = false;
+		
+		if ((currDropzone)&&(currDropzone.type == 'final')) {
+			isWin();
+		}
+
+	}
+
+	function placeCard(card, dropzone) {
+		
+		if (card.placedOn) {
+			card.placedOn.cardsOnTop.length--;
+		}
+			
+		card.elem.style.zIndex = dropzone.cardsOnTop.length + 1;
+		card.elem.style.left = dropzone.elem.style.left;
+				
+		if (dropzone.type == 'temp') {
+			if (dropzone.cardsOnTop.length == 0) {
+				card.elem.style.top = parseFloat(dropzone.elem.style.top) + 'px';
+			} else if (dropzone.cardsOnTop[dropzone.cardsOnTop.length - 1].visibility == 'opened') {
+				card.elem.style.top = parseFloat(dropzone.cardsOnTop[dropzone.cardsOnTop.length - 1].elem.style.top) + 20 + 'px';
+			} else {
+				card.elem.style.top = parseFloat(dropzone.cardsOnTop[dropzone.cardsOnTop.length - 1].elem.style.top) + 5 + 'px';
+			}
+		} else {
+			card.elem.style.top = parseFloat(dropzone.elem.style.top) + 'px';
+		}
+		
+		dropzone.cardsOnTop.push(card);
+		card.placedOn = dropzone;
+		
+	}
+
+	function checkCardColor(card) {
+		
+		switch (card.colour) {
+			
+			case 'diamonds':
+			case 'hearts':
+				return 'red';
+				
+			case 'spades':
+			case 'clubs':
+				return 'black';
+				
+		}
+		
+	}
+
+	function dropRuleCheck(card, dropzone) {
+		
+		switch (dropzone.type) {
```

Comment:
you can make an object with name-filter function 
`
var filters = {
  temp: function(current card) {
 return true/false;
  }
};


var ableToDrop = filters[dropzone.type](some parameters);

`